### ISSUE TITLE
bug: Controller executing twice - nested states.

### ISSUE THREAD

**Issue by Leftyx** [driftyco/ionic#4089]
**Type**: <span ionic-type>bug</span>

**Platform**: <span ionic-platform>all</span>  

<span ionic-description>Nested states with parameters and tabs execute the controller twice.

How to reproduce:

The first state manages the tabs 

```
  .state('tabs', {
      url: "/tab/:projectID",
      cache: false,
      templateUrl: "tabs.html",
      controller: "TabsCtrl  as vm",
      resolve: {
        project: function($stateParams, Projects) {
          return Projects.get($stateParams.projectID);
        }
      }
  })

  .state('tabs.tab1', {
      url: "/tab1",
      views: {
          'tabs-tab1': {
              templateUrl: "tab1.html",
              controller: 'Tab1Ctrl as vm'
          }
      }
  })

```
the seconds loads the specific tabs in the tab list.

Plunker [here](http://plnkr.co/edit/3al3x1OVlZ1qXabWz36H?p=preview).

Question asked on [StackOverflow](http://stackoverflow.com/questions/31392469/controller-executing-twice-with-ui-router-nested-states-ionic).
</span>

<span is-issue-template></span>

**Comment by Leftyx** [comment]
It seems that this [line](https://github.com/driftyco/ionic/blob/6c9bc15b2996ae9183624c84a2709fd33f80d20b/js/angular/controller/navViewController.js#L269) calls the controller twice.

**Comment by Ionitron** [comment]
Greetings @Leftyx!

I've closed this issue because my sensors indicated it was old and inactive, and may have already been fixed in recent versions of Ionic. However, if you are still experiencing this issue, please feel free to reopen this issue by creating a new one, and include any examples and other necessary information, so that we can look into it further.

Thank you for allowing me to assist you.

**Comment by kylefox** [comment]
FWIW the [original Plunker](http://plnkr.co/edit/3al3x1OVlZ1qXabWz36H?p=preview) appears to still demonstrate the problem (Ionic, v1.1.0-nightly-1583). Perhaps related to nested `<ion-nav-view></ion-nav-view>` elements?

If nobody else has been bitten by this then maybe it's not worth addressing or opening a new issue for.

**Comment by rakesh1988** [comment]
I hit the same route today.. I could see that my controller was called twice when I set cache false on one of the page. Is this something to worry about?

**Comment by queejie** [comment]
Yes, it is causing data to load twice for me in the controller.  Oddly enough, if I put a check in the service that loads the data to make sure it doesn't load if already started, the data change doesn't get noticed in the scope.  If I remove the abstract parent (for ionic tabs) it works fine.

**Comment by peterzebird** [comment]
+1

**Comment by charris-RA** [comment]
+1

**Comment by lesserenterpriser** [comment]
I'm having a similar problem, but for me the controllers are being called more than twice.  I'm tracking it with simple console.log's when the state is called (e.g. console.log('Tab1.child is active').  I'll see that message sometimes 2 times, sometimes 4 times, sometimes 8, sometimes 10.  Not sure what's going on, but definitely seems to affect performance.  Any forward movement on this issue?

**Comment by slumrig** [comment]
+1 
I'm hitting this issue with ionic version 1.7.15. To get around it should I dispense with having nested views? I'm trying to build the view structure in a well-composed way but this issue is really putting a damper on that.

**Comment by matiu** [comment]
+1,

Ionic 1.3.1.


The controller fires +1 times every time I enter to it.....

**Comment by martyzz1** [comment]
I'm observing this also. I have 4 tabs, and each tab controller loads 4 times.

**Comment by ASHBAW** [comment]
I am also seeing the same thing as queejie.  Any solution to this yet?

**Comment by veljkocasa** [comment]
+1 / ionic 1.3.0 
One instance more on each invocation! with and without caching ...
Does anyone found out where is a problem, do we do something wrong??

So long with no answer, where is the ionic team

**Comment by veljkocasa** [comment]
@matiu , I suppose you've found some way.... can you tell, pls?

**Comment by eduzera** [comment]
+1

**Comment by jachstet-sea** [comment]
I am experiencing exactly the same problem and any workaround like constructing some kind of singleton leads to even more headache. @Ionitron, @Leftyx or @kylefox  can this issue please be re-opened?

**Comment by jachstet-sea** [comment]
Oh, and to not come here empty-handed: In the application where I have the same problem, I added a console.trace() to the controller's constructor. The interesting thing is that the trace is different for the two calls. So I added the console.trace() to the plunkr as well: http://plnker.co/edit/xJ6WBLubrx4bCbJmCqNE?p=preview
The one trace contains the lines:
`$RootScopeProvider/this.$get</Scope.prototype.$broadcast()ionic.bundle.js:30717
transitionTo/$state.transition<()`
The other one these lines:
`publicLinkFn()ionic.bundle.js:21582
tabSelected()`

So while the one constructor is directly triggered by the switching of the tabs, the other one seems to be triggered by the URL/state change which itself is triggered by the tab click.

Maybe one of those events should be prevented when clicking on a tab?
